pyrovider/services/provider.py

```diff
@@ -81,17 +148,78 @@ def _init_local(self):
     def reset(self):
         release_local(self._local)
 
+        for p in self._providers:
+            p.reset()
+
     def conf(self, service_conf: dict, app_conf: dict = None):
```

Comment:
Why do we need this method now that the provider isn't a singleton anymore? Could this not just be in the init?

Reply:
this was done to make sure the state of a provider is not persisted across different requests, so on request teardown we can call reset.